src/main/groovy/io/cosmicsilence/scalafix/ScalafixPlugin.groovy

```diff
@@ -23,18 +28,19 @@ class ScalafixPlugin implements Plugin<Project> {
 
     @Override
     void apply(Project project) {
+        if (!project.plugins.hasPlugin(ScalaPlugin)) {
+            throw new GradleException("'scala' plugin must be applied")
+        }
```

Comment:
I'm tempted to break the rules and just apply the Scala plugin here. This check requires the `scala` plugin to be applied prior to the `scalafix` one. What do you think is worse? The other option is to keep as is and silently skip the application of our plugin (maybe log something)

Reply:
I consider scala-fix an addon to a scala project, therefore applying scala before scala-fix would be common sense. Also, if I had to go through a `build.gradle` file of a project, it looks clearer to me if I see both are applied.

Personally, I prefer to throw an exception if you're applying it to a project where the scala plugin hasn't been applied.

Reply:
ok, you convinced me 😄